feeratekit/src/main/kotlin/io/horizontalsystems/feeratekit/model/FeeRate.kt

```diff
@@ -1,82 +1,80 @@
 package io.horizontalsystems.feeratekit.model
 
-import androidx.room.Entity
-import androidx.room.PrimaryKey
-import androidx.room.TypeConverters
-import io.horizontalsystems.feeratekit.storage.EnumConverter
+import java.util.*
 
-@Entity
-@TypeConverters(EnumConverter::class)
 
 data class FeeRate(
-    @PrimaryKey
-    val coin: Coin,
-    val lowPriority: Long,
-    val lowPriorityDuration: Long,
-    val mediumPriority: Long,
-    val mediumPriorityDuration: Long,
-    val highPriority: Long,
-    val highPriorityDuration: Long,
-    val date: Long
-) {
+        val coin: Coin,
+        val lowPriority: Long,
+        val lowPriorityDuration: Long,
+        val mediumPriority: Long,
+        val mediumPriorityDuration: Long,
+        val highPriority: Long,
+        val highPriorityDuration: Long,
+        val date: Long) {
+
     fun safeLow(): Long = Math.max(coin.minRate(), Math.min(lowPriority, coin.maxRate()))
     fun safeMedium(): Long = Math.max(coin.minRate(), Math.min(mediumPriority, coin.maxRate()))
     fun safeHigh(): Long = Math.max(coin.minRate(), Math.min(highPriority, coin.maxRate()))
+
 }
 
-enum class Coin(val code: String) {
-    BITCOIN("BTC"),
-    BITCOIN_CASH("BCH"),
-    DASH("DASH"),
-    ETHEREUM("ETH");
+enum class Coin(
+        val code: String,
+        val fallbackDataExpiration: Int) {
```

Comment:
I think better place for expiration time would be in the BaseFeeRateProvider. Coin knowing about expiration time of FeeRate looks strange to me.